src/main/webapp/app/exercises/text/assess-new/text-submission-assessment.route.ts

```diff
@@ -12,39 +12,44 @@ import { TextFeedbackConflictsComponent } from './conflicts/text-feedback-confli
 import { Authority } from 'app/shared/constants/authority.constants';
 
 @Injectable({ providedIn: 'root' })
-export class StudentParticipationResolver implements Resolve<StudentParticipation | undefined> {
-    constructor(private textAssessmentsService: TextAssessmentsService) {}
+export class NewStudentParticipationResolver implements Resolve<StudentParticipation | undefined> {
+    constructor(private textSubmissionService: TextSubmissionService) {}
 
     /**
      * Resolves the needed StudentParticipations for the TextSubmissionAssessmentComponent using the TextAssessmentsService.
      * @param route
      */
     resolve(route: ActivatedRouteSnapshot) {
-        const submissionId = Number(route.paramMap.get('submissionId'));
-
-        if (submissionId) {
-            return this.textAssessmentsService.getFeedbackDataForExerciseSubmission(submissionId).catch(() => Observable.of(undefined));
+        const exerciseId = Number(route.paramMap.get('exerciseId'));
+        console.log('router: resolve:');
+        const correctionRound = Number(route.queryParamMap.get('correction-round'));
+        console.log('correctionRound: ', correctionRound);
+        if (exerciseId) {
```

Comment:
This should also check if `correctionRound` is valid (not `NaN` for example).

Reply:
Are you sure we need to check it here? It is also not checked for all the other parameters, and if it really turns out to be a NaN then the thrown Exception will be catched.

Reply:
> It is also not checked for all the other parameters

I might be missing something, but the only parameters used here are `exerciseId` and `correctionRound`? (Well, `'lock'` but that can't be wrong since it is a constant).
It would indeed catch the exception (so in the end this doesn't matter too much), but we're still sending a request to the server we know will fail for certain.

Reply:
Actually on a deeper look this should be handled in `textSubmissionService` which would ignore this error.